src/lib/models/transaction.js

```diff
@@ -185,9 +178,60 @@ class Transaction extends Model {
   }
 
   set volume (volume) {
-    assertQuantity(volume);
+    validators.assertQuantity(volume);
     this._volume = volume;
   }
+
+  /**
+   * Creates a POJO containing a single layer of data that will be
+   * used to create a unique hash for this transaction
+   *
+   * @param {String} accountNumber The account number
+   * @param {Licence} licence Licence information
+   * @param {Region} region Region object
+   */
+  getHashData (accountNumber, licence) {
+    return {
+      periodStart: this.chargePeriod.startDate,
+      periodEnd: this.chargePeriod.endDate,
+      billableDays: this.billableDays,
+      authorisedDays: this.authorisedDays,
+      volume: this.volume,
+      agreements: this.agreements.map(ag => ag.code).sort().join('-'),
+      accountNumber,
+      source: this.chargeElement.source,
+      season: this.chargeElement.season,
+      loss: this.chargeElement.loss,
+      description: this.description,
+      licenceNumber: licence.licenceNumber,
+      regionCode: licence.region.code
```

Comment:
Comparing this with what is sent to the CM, is it still missing a few properties, such as:
* type (compensation|standard)
* twoPartTariff (this is taken from the containing batch and is distinct from whether a TPT agreement applies).